compiler/crates/relay-lsp/src/server/lsp_request_dispatch.rs

```diff
@@ -79,6 +80,14 @@ fn convert_to_lsp_response(
             error: None,
         },
         Err(runtime_error) => {
+            if let LSPRuntimeError::ExpectedError = runtime_error {
+                return ServerResponse {
+                    id,
+                    result: Some(Value::Null),
+                    error: None,
+                };
+            }
+
```

Comment:
Hmm, why do we need to change this? Is this related to the path?

Reply:
Wupps, just mentioned that in the original issue, but not the PR description 😅

If the file is not part of the directory the LSP operates on (as checked by the various ifs I made changes to), the methods return a `LSPRuntimeError::ExpectedError`. The `LSPRuntimeError` is converted to a `ResponseError` here: https://github.com/facebook/relay/blob/main/compiler/crates/relay-lsp/src/server/lsp_request_dispatch.rs#L82

In this conversion a `ExpectedError` will be mapped to `None`: https://github.com/facebook/relay/blob/main/compiler/crates/relay-lsp/src/lsp_runtime_error.rs#L22

The `None` is mapped to a `Request cancelled` error as a fallback I supose: https://github.com/facebook/relay/blob/main/compiler/crates/relay-lsp/src/server/lsp_request_dispatch.rs#L83-L87

The `Request cancelled` error sent me on a totally wrong track to tackle the underlying issue at first, since I thought the request to the LSP was being aborted only on Windows... So I opted to not return an error in the case of an expected error. I didn't find anything about how to handle expected errors in an LSP, so I just opted to return a `null` result. I wasn't sure how to handle it better as I'm a novice both regarding LSPs and Rust. Maybe one of you guys has a better idea how to surface expected errors correctly.

Reply:
Done :)